// Styled/Components.js
import { styled } from 'Styled'

const PageWrapper = styled('div', {
  boxSizing: 'border-box',
  display: 'flex',
  flexDirection: 'column',
  maxHeight: '100vh',
  overflow: 'hidden'
})
const PageHeader = styled('header', {
  padding: "20px 40px"
})
const PageContent = styled('div', {
  overflow: 'scroll',
  padding: "0 40px"
})
export const Page = ({ title, header, children }) => (
  <PageWrapper>
    <PageHeader>
      <Title>{title}</Title>
      {header}
    </PageHeader>
    <PageContent>
      {children}
    </PageContent>
  </PageWrapper>
)

export const Title = styled("h1", {
  margin: 0,
  paddingBottom: 10
})

export const BigButton = styled('button', {
  fontWeight: 'bold',
  background: '#1976d2',
  color: 'white',
  borderRadius: 7,
  border: 'none',
  outline: 'none',
  padding: '10px 20px',
  textTransform: 'uppercase',
  marginRight: 10,
  cursor: 'pointer',
  ":disabled": {
    opacity: 0.4,
    cursor: 'not-allowed'
  },
  ":hover:not(:disabled)": {
    boxShadow: '0 3px #0051a0',
    transform: 'translateY(-3px)'
  },
  ":active:not(:disabled)": {
    boxShadow: 'none',
    transform: 'translateY(0)'
  }
})
